src/request/xhr.ts

```diff
@@ -67,8 +69,15 @@ export default function xhr<T>(url: string, options: XhrRequestOptions = {}): Re
 
 				response.statusCode = request.status;
 				response.statusText = request.statusText;
-
-				resolve(response);
+				if (response.statusCode >= 200 && response.statusCode < 400) {
```

Comment:
This ties into your comment on the 'timeout' test that's failing. When the request times out, prior to the ontimeout callback getting fired, this code gets fired with a status code of 0. I looked into it and apparently 0 is  a valid value for XMLHttpRequest's statusCode property, indicating an error. I could change this to `response.statusCode < 400 && response.statusCode !== 0`, but I'm still not sure why this is getting called with a statusCode of 0, and how we want to handle that with regards to the 'timeout' test.